src/config/Icon.tsx

```diff
@@ -115,12 +118,19 @@ const NamedIcon = ({
   color = colors.primary,
   iconStyle,
   name,
+  stroke = 1,
   size = normalize(26),
   style
-}: IconProps & { name?: ComponentProps<typeof IconSet>['name'] }) => {
+}: IconProps & {
+  name?: ComponentProps<typeof IconSet>['name'];
+  stroke?: ComponentProps<number>;
+}) => {
+  const TablerName = ('Icon' + _upperFirst(_camelCase(name))) as keyof typeof Tabler;
+  const IconComponent = Tabler?.[TablerName] || IconSet;
+
   return (
     <View accessibilityLabel={accessibilityLabel} style={style} hitSlop={getHitSlops(size)}>
-      <IconSet name={name} size={size} color={color} style={iconStyle} />
+      <IconComponent name={name} size={size} color={color} style={iconStyle} stroke={stroke} />
```

Comment:
so let's summarize, we pass a `name` to `NamedIcon` without knowing if it is from `IconSet` or `Tabler`? `name` is marked as optional with ?, wich is not possible, right? we need a name! and the type is not of `IconSet` anymore but from `IconSet` or `Tabler`, so i guess the types are not final right now.

what happens, if we pass a `name` that exists for `Tabler` and `IconSet` and we want it from `IconSet`? with the current logic, `Tabler` would be used first.

i think we need a more detailed separation or test for the two possible icon types, before setting the one used as `IconComponent`.

Reply:
Okay, you're right. How about adding a control like this before the return? That way we can check which `IconSet` we are using and create the `IconComponent` accordingly.


```
  let IconComponent: any;

  if (IconSet === Tabler) {
    const TablerName = ('Icon' + _upperFirst(_camelCase(name))) as TablerIconName;
    IconComponent = Tabler[TablerName] || Tabler['IconQuestionMark'];
  } else {
    IconComponent = IconSet;
  }

  return (
    <View accessibilityLabel={accessibilityLabel} style={style} hitSlop={getHitSlops(size)}>
      <IconComponent name={name} size={size} color={color} style={iconStyle} stroke={stroke} />
    </View>
  );
```

Reply:
the code makes much more sense this way, yes! so we clearly have one type of icon set and not a mix of two. great!
please change it and i will re-check.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/83142b71005d5e1a0824d8e710af96c5f6d18453